lib/LedgerSMB/Report/Balance_Sheet.pm

```diff
@@ -255,11 +255,10 @@ sub run_report {
             my $cat = $props->{account_category};
             my $contra = $props->{contra};
 
-            my $sign = (($contra) ? -1 : 1)
-                * ((($cat eq 'A') || ($cat eq 'E')) ? -1 : 1);
+            my $is_positive = !!$contra == ( $cat eq 'A' or $cat eq 'E' );
 
-            $self->cell_value($row_id, $col_id, $sign * $value)
-                if $sign < 0;
+            $self->cell_value($row_id, $col_id, -$value)
+                unless $is_positive;
         }
```

Comment:
There is not any issue.   I didn't mean to put you to work, I am sorry it I did.